rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -310,15 +318,17 @@ impl TunnelManager {
         let iface_index = if let Some(dev) = dev.clone() {
             KI.get_iface_index(&dev).unwrap()
         } else {
-            0
+            return Box::new(futures::future::err(TunnelManagerError::NoDevError.into()));
```

Comment:
What does dev "none" normally do exactly? does it mean it takes the default route?

Reply:
0 means do whatever the kernel routing table tells us to do, instead of specifying a specific interface (should not be an error)

Reply:
Why do we take `dev` as an `Option` here if we're not going to handle a `None` gracefuly?